gpt_index/vector_stores/pinecone.py

```diff
@@ -39,6 +43,43 @@ def get_node_info_from_metadata(
     return node_extra_info
 
 
+def build_dict(input_batch: List[List[int]]) -> List[Dict[str, Any]]:
+    """Build a list of sparse dictionaries from a batch of input_ids.
+
+    NOTE: taken from https://www.pinecone.io/learn/hybrid-search-intro/.
+
+    """
+    # store a batch of sparse embeddings
+    sparse_emb = []
+    # iterate through input batch
+    for token_ids in input_batch:
+        indices = []
+        values = []
+        # convert the input_ids list to a dictionary of key to frequency values
+        d = dict(Counter(token_ids))
+        for idx in d:
+            indices.append(idx)
+            values.append(float(d[idx]))
+        sparse_emb.append({"indices": indices, "values": values})
+    # return sparse_emb list
+    return sparse_emb
+
+
+def generate_sparse_vectors(
```

Comment:
related to comment below. Is this specific to Pinecone or could it be moved to a more general scope?

Reply:
this was copied from their tutorial - i think it's specific to pinecone

different vector stores have different interfaces for hybrid search e.g. weaviate just has you pass in the raw text, so i didn't want to create sparse vector outside